src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/casedocuments/PartyDocumentHandler.java

```diff
@@ -0,0 +1,49 @@
+package uk.gov.hmcts.reform.finrem.caseorchestration.service.casedocuments;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.ContestedUploadedDocumentData;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Slf4j
+public abstract class PartyDocumentHandler extends CaseDocumentHandler {
+
+    private final String collectionName;
+    private final String party;
+    private final ObjectMapper mapper;
+
+    protected static final String APPLICANT = "applicant";
+    protected static final String RESPONDENT = "respondent";
+
+    public PartyDocumentHandler(String collectionName, String party, ObjectMapper mapper) {
+        super(mapper);
+        this.collectionName = collectionName;
+        this.party = party;
+        this.mapper = mapper;
+    }
+
+    public void handle(List<ContestedUploadedDocumentData> uploadedDocuments,
+                       Map<String, Object> caseData) {
+        List<ContestedUploadedDocumentData> documentsFiltered = uploadedDocuments.stream()
+            .filter(d -> d.getUploadedCaseDocument().getCaseDocuments() != null
```

Comment:
seems like these are null checked elsewhere too so could probs be extracted to a public method